packages/go/analysis/ad/queries.go

```diff
@@ -1471,34 +1472,91 @@ func FetchCertTemplatesPublishedToCA(tx graph.Transaction, ca *graph.Node) (grap
 	}))
 }
 
-func FetchCanAbuseWeakCertBindingRels(tx graph.Transaction, node *graph.Node) ([]*graph.Relationship, error) {
-	if rels, err := ops.FetchRelationships(tx.Relationships().Filterf(func() graph.Criteria {
-		return query.And(
-			query.Equals(query.StartID(), node.ID),
-			query.Kind(query.Relationship(), ad.CanAbuseWeakCertBinding),
-			query.Kind(query.End(), ad.Entity),
-		)
-	})); err != nil {
-		return nil, err
+func HasUPNCertMappingInForest(tx graph.Transaction, domain *graph.Node) (bool, error) {
```

Comment:
Would it be possible to move the usage of this function and the `HasWeakCertBindingInForest` up the stack into ADCSCache's `BuildCache` method instead of within `PostADCSXy` functions? This would reduce the amount of times that we crawl domain trusts and query the db since the `PostADCSXy` functions are nested in a double loop going through all domains and all enterpriseca's.

Reply:
That is a good idea. I will take a look at that 👍